tests/unit/test_charm.py

```diff
@@ -524,6 +525,22 @@ def test_postgresql_layer(self, _, __, ___):
                         "PATRONI_SUPERUSER_USERNAME": "operator",
                     },
                 },
+                self._metrics_service: {
+                    "override": "replace",
+                    "summary": "postgresql metrics exporter",
+                    "command": "/start-exporter.sh",
+                    "startup": "enabled",
+                    "after": [self._postgresql_service],
+                    "user": "postgres",
+                    "group": "postgres",
+                    "environment": {
+                        "DATA_SOURCE_NAME": (
+                            f"user=monitoring "
+                            f"password={self.charm.get_secret('app', 'monitoring-password')} "
+                            "host=/var/run/postgresql port=5432 database=postgres"
```

Comment:
Should you use constant DATABASE_PORT instead of hardcoded port?

Reply:
the exporter is using a socket connection, but apparently validates if a port is defined in the envvar, independently of the value